packages/plugin_platform_interface/test/plugin_platform_interface_test.dart

```diff
@@ -21,6 +21,12 @@ class SamplePluginPlatform extends PlatformInterface {
 class ImplementsSamplePluginPlatform extends Mock
     implements SamplePluginPlatform {}
 
+class ImplementsSamplePluginPlatformUsingNoSuchMethod
+    implements SamplePluginPlatform {
+  @override
+  dynamic noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
```

Comment:
Should this implementation maybe be a no-op instead of a call to `super` so we're validating that someone can't completely break the check with a non-standard `noSuchMethod`?

Reply:
That might be an interesting test case to add, but it's not the test case that's fixed by this PR.  Even before this change, a `noSuchMethod` that was a no-op (and returned, say `null`) would fail the `if (!identical(token, instance._instanceToken))` check, and cause `_verify` to throw an `AssertionError`, as intended.

Reply:
Ah, right. Sorry, the exact details here are subtle so they get swapped out pretty quickly for me :)